pkgs/applications/audio/musescore/default.nix

```
@@ -1,65 +1,146 @@
-{ mkDerivation, lib, fetchFromGitHub, fetchpatch, cmake, pkg-config, ninja
-, alsa-lib, freetype, libjack2, lame, libogg, libpulseaudio, libsndfile, libvorbis
-, portaudio, portmidi, qtbase, qtdeclarative, qtgraphicaleffects, flac
-, qtquickcontrols2, qtscript, qtsvg, qttools
-, qtwebengine, qtxmlpatterns, qtnetworkauth, qtx11extras
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchpatch
+, cmake
+, wrapQtAppsHook
+, pkg-config
+, ninja
+, alsa-lib
+, freetype
+, libjack2
+, lame
+, libogg
+, libpulseaudio
+, libsndfile
+, libvorbis
+, portaudio
+, portmidi
+, qtbase
+, qtdeclarative
+, qtgraphicaleffects
+, flac
+, qtquickcontrols
+, qtquickcontrols2
+, qtscript
+, qtsvg
+, qtxmlpatterns
+, qtnetworkauth
+, qtx11extras
 , nixosTests
+, darwin
 }:
 
-mkDerivation rec {
+let
+  stdenv' = if stdenv.isDarwin then darwin.apple_sdk_11_0.stdenv else stdenv;
```

```suggestion
  stdenv' = darwin.apple_sdk_11_0.stdenv;
```
the implementation is already doing this

---

> the implementation is already doing this

I thought so too but I wasn't sure. Thanks!

---

Wait no, I get "unsupported platform" errors with this change, because `darwin.apple_sdk_11_0.stdenv.isDarwin` is always `true`.

---

Do you need the `stdenv'` at all if the package is `apple_sdk_11_0.callPackage`?

---

> Do you need the `stdenv'` at all if the package is `apple_sdk_11_0.callPackage`?

The `portaudio` package is not defined using `darwin.apple_sdk_11_0.callPackage` if that's what you meant. How these things should be composed is discussed in general in https://github.com/NixOS/nixpkgs/issues/242666 .

---

> Do you need the `stdenv'` at all if the package is `apple_sdk_11_0.callPackage`?

portaudio uses Qt’s callPackage, which doens’t appear to provide the 11.0 SDK stdenv. The Qt `callPackage` is needed to make sure the correct Qt dependencies are provided to the package.

> the implementation is already doing this

Only if you use `darwin.apple_sdk_11_0.callPackage`. The 11.0 SDK stdenv itself is not defined conditionally (except to pass through the existing 11.0 SDK stdenv on aarch64-darwin).